src/flatland/ordered/map.clj

```diff
@@ -86,7 +86,8 @@
     (.equiv this other))
   (hashCode [this]
     (reduce (fn [acc ^MapEntry e]
-              (let [k (.key e), v (.val e)]
+              (let [k (if-some [k (.key e)] k 0)
```

Comment:
I'm not totally clear on what cases use hashCode and which use hasheq instead, but it's best to be the same for both - you can see that the original implementation of hashCode was an attempt to transliterate the existing java code (for clojure versions before it was exposed as a public static helper method).